tests/unit/core/server/test_base.js

```diff
@@ -222,6 +225,106 @@ describe(__filename, () => {
+
+      const dispatchSpy = sinon.spy(store, 'dispatch');
+
+      const response = await testClient({
+        appInstanceName: 'disco',
+        config: fakeConfig,
+        store,
+        sagaMiddleware,
+      })
+        .get('/en-US/firefox/')
+        .set('cookie', `${fakeConfig.get('discoTaarIdCookie')}="${clientId}"`)
+        .end();
+
+      sinon.assert.calledWith(dispatchSpy, setHashedClientId(clientId));
+
+      expect(response.statusCode).toEqual(200);
+      expect(store.getState().telemetry.hashedClientId).toEqual(clientId);
+    });
+
+    it('does not dispatch setHashedClientId() if enableFeatureDiscoTaar is false', async () => {
+      const clientId = '1112';
+      const fakeConfig = getFakeConfig({
+        enableFeatureDiscoTaar: false,
+      });
+
+      const { store, sagaMiddleware } = createStoreAndSagas();
+
+      const dispatchSpy = sinon.spy(store, 'dispatch');
+
+      const response = await testClient({
+        appInstanceName: 'disco',
+        config: fakeConfig,
+        store,
+        sagaMiddleware,
+      })
+        .get('/en-US/firefox/')
+        .set('cookie', `${fakeConfig.get('discoTaarIdCookie')}="${clientId}"`)
+        .end();
+
+      sinon.assert.neverCalledWith(dispatchSpy, setHashedClientId(clientId));
+
+      expect(response.statusCode).toEqual(200);
+      expect(store.getState().telemetry.hashedClientId).toEqual(null);
+    });
+
+    it('does not dispatch setHashedClientId() if there is no discoTaarIdCookie', async () => {
+      const fakeConfig = getFakeConfig({
+        enableFeatureDiscoTaar: true,
+      });
+      const { store, sagaMiddleware } = createStoreAndSagas();
+
+      const dispatchSpy = sinon.spy(store, 'dispatch');
+
+      const response = await testClient({
+        appInstanceName: 'disco',
+        config: fakeConfig,
+        store,
+        sagaMiddleware,
+      })
+        .get('/en-US/firefox/')
+        .end();
+
+      sinon.assert.neverCalledWith(dispatchSpy, setHashedClientId('1112'));
+
+      expect(response.statusCode).toEqual(200);
+      expect(store.getState().telemetry.hashedClientId).toEqual(null);
+    });
+
+    it('does not dispatch setHashedClientId() if app is not disco', async () => {
+      const clientId = '1112';
+      const fakeConfig = getFakeConfig({
+        enableFeatureDiscoTaar: true,
+      });
+      const { store, sagaMiddleware } = createStoreAndSagas();
+
+      const dispatchSpy = sinon.spy(store, 'dispatch');
+
+      const response = await testClient({
+        appInstanceName: 'not-disco',
+        config: fakeConfig,
+        store,
+        sagaMiddleware,
+      })
+        .get('/en-US/firefox/')
+        .set('cookie', `${fakeConfig.get('discoTaarIdCookie')}="${clientId}"`)
+        .end();
+
+      sinon.assert.neverCalledWith(dispatchSpy, setHashedClientId(clientId));
+
+      expect(response.statusCode).toEqual(200);
+      expect(store.getState().telemetry.hashedClientId).toEqual(null);
```

Comment:
Is this possible? We should not have `telemetry` in the store when app is `disco`.

Reply:
yeh good catch. I think I need to adjust the logic a bit.

Reply:
telemetry *should* only be in the store when it IS disco..